File: 1116-maximum-level-sum-of-a-binary-tree/1116-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        q = [root]
        max_lvl = 1
        lvl = 1
        max_sum = -(2 ** 32)
        while len(q) != 0:
            size = len(q)
            lvl_sum = 0
            for i in range(size):
                top = q.pop(0)
                lvl_sum += top.val
                if top.left:
                    q.append(top.left)
                if top.right:
                    q.append(top.right)
            if lvl_sum > max_sum:
                max_sum = lvl_sum
                max_lvl = lvl
            lvl += 1
        return max_lvl

File: 1116-maximum-level-sum-of-a-binary-tree/1116-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        std::queue<TreeNode*> q;
        q.push(root);
        int max_lvl = 1;
        int lvl = 1;
        int max_sum = INT_MIN;
        while (!q.empty()) {
            int level_sum = 0;
            int size = q.size();
            for (int i = 0; i < size; ++i) {
                auto top = q.front();
                q.pop();
                level_sum += top->val;
                if (top->left) {
                    q.push(top->left);
                }
                if (top->right) {
                    q.push(top->right);
                }
            }
            if (level_sum > max_sum) {
                max_sum = level_sum;
                max_lvl = lvl;
            }

            ++lvl;
        }
        return max_lvl;
    }
};